sklearn/tree/export.py

```diff
@@ -216,6 +217,19 @@ def node_to_str(tree, node_id, criterion):
         if tree.children_left[node_id] != _tree.TREE_LEAF:
             # Always write node decision criteria, except for leaves
             if feature_names is not None:
+                # Check length of feature_names
+                # raise error for too few feature_names
+                if len(feature_names) < tree.n_features:
```

Comment:
Failing early is always better.

In other words, you should check this once at the beginning when `export_graphviz` is called rather than when `node_to_str` is called. You can do that by accessing the attribute `n_features_` of the provided `decision_tree`.